tests/unit/test_network_connections.py

```diff
@@ -4081,6 +4095,145 @@ def test_convert_passwd_flags_nm(self):
+        val_min = 1
+        val_max = 0xFFFFFFFF
+        self.assertRaisesRegex(
+            ValidationError,
+            "route table value is %s but cannot be less then %s"
+            % (
+                self.test_connections[0]["ip"]["route"][0]["table"],
+                val_min,
+            ),
+            self.validator.validate,
+            self.test_connections,
+        )
+
+        self.test_connections[0]["ip"]["route"][0]["table"] = 4294967296
+        self.assertRaisesRegex(
+            ValidationError,
+            "route table value is %s but cannot be greater then %s"
+            % (
+                self.test_connections[0]["ip"]["route"][0]["table"],
+                val_max,
+            ),
+            self.validator.validate,
+            self.test_connections,
+        )
+
+    def test_invalid_named_route_tables(self):
+        """
+        Test that the validation error will be raised when the named route table is not
+        found in the `/etc/iproute2/rt_tables`
+        """
+
+        self.test_connections[0]["ip"]["route"][0]["table"] = "eth"
+
+        self.assertRaisesRegex(
+            ValidationError,
+            "can not find route table '%s' in `/etc/iproute2/rt_tables`"
+            % (self.test_connections[0]["ip"]["route"][0]["table"]),
+            self.validator.validate_route_tables,
+            self.validator.validate(self.test_connections)[0],
+            self.connection_index,
+        )
+
+    def test_valid_named_route_tables(self):
+        """
+        Test that the validation error will not be raised when the named route table is
+        found in the `/etc/iproute2/rt_tables`, and such a named route table will be
+        normalized into the corresponding route table id.
+        """
+
+        self.test_connections[0]["ip"]["route"][0]["table"] = "main"
+        self.validator.validate_route_tables(
+            self.test_connections[0], self.connection_index
+        )
+        self.assertEqual(self.test_connections[0]["ip"]["route"][0]["table"], 254)
+
+    def test_get_route_tables_mapping(self):
+        """
+        Test that `RouteTableParsing.get_route_tables_mapping()` will parse the files
+        `/etc/iproute2/rt_tables` and return a dictionary containing the mapping
+        between the route table name and route table id.
+        """
+        cached_rt_tables = self.rt_parsing.get_route_tables_mapping()
+        self.assertEqual(
+            cached_rt_tables, {"local": "255", "main": "254", "default": "253"}
+        )
+
+    def test_parse_route_tables_mapping(self):
+        """
+        Test that `RouteTableParsing.parse_route_tables_mapping()` will parse the bytes
+        file content in `/etc/iproute2/rt_tables` and return a dictionary containing
+        the mapping between the route table name and route table id.
+        """
+        rt_file = "/etc/iproute2/rt_tables"
+        with open(rt_file, "r") as f:
```

Comment:
the unit test shouldn't read files from disk on the system where the unit test runs.

Here, you are not even testing `parse_route_tables_mapping_from_file()`, but `parse_route_tables_mapping()`. Just hard code a large set of input "file_contents" that you test in the test here.

If you would add a test for `parse_route_tables_mapping_from_file()`, then you would need to have a file on disk. But that file must be one shipped by the unit tests itself. But `parse_route_tables_mapping_from_file()` is rather trivial. It doesn't seem necessary to test the "can read file" part.

-- at which point you might ask, why does  `parse_route_tables_mapping_from_file()` even exist as separate function. You could drop it and inline all the logic into `get_route_tables_mapping()`. But both is fine.

Reply:
I dropped the unit tests here.

Reply:
I dropped the unit tests here.

Reply:
And I also dropped the `parse_route_tables_mapping_from_file()`.